buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy

```diff
@@ -115,6 +115,11 @@ class PluginBuildPlugin implements Plugin<Project> {
             if (isModule == false || isXPackModule) {
                 addNoticeGeneration(project, extension1)
             }
+            if (project.pluginManager.hasPlugin("elasticsearch.yaml-rest-test")
```

Comment:
Let's not use `afterEvaluate` + `hasPlugin`. This is subject to configuration ordering. Let's instead move this outside of `afterEvaluate` and just use `withPlugin` to disable that `integTest` task whenever either of these plugins is applied.